src/pudl/output/ferc1.py

```diff
@@ -871,3 +873,381 @@ def add_mean_cap_additions(steam_df):
+    ]
+    logger.info(
+        f"Explode: Preparing to drop: {inter_table_facts_to_drop.xbrl_factoid_name_original}"
+    )
+    # TODO: We need to fill in the other_dimensions columns before doing this check.
+    # check to see if there are different values in the values that show up in two tables
+    inter_table_ref_check = (
+        # these are all the dupes not just the ones we are axing
+        exploded[
+            exploded.xbrl_factoid_name_original.isin(
+                inter_table_facts_to_drop.xbrl_factoid_name_original
+            )
+        ]
+        .groupby(
+            [col for col in pks if col != "xbrl_factoid"]
+            + ["xbrl_factoid_name_original"],
+            dropna=False,
+        )[[value_col]]
+        .nunique()
+    )
+    assert inter_table_ref_check[inter_table_ref_check[value_col] > 1].empty
+
+    factoid_idx = ["xbrl_factoid_name_original", "table_name"]
+    exploded = exploded.set_index(factoid_idx)
+    inter_table_refs = exploded.loc[
+        inter_table_facts_to_drop.set_index(factoid_idx).index
+    ]
+    if len(inter_table_refs) > 1:
+        logger.info(
+            f"Explode: Dropping {len(inter_table_refs)} ({len(inter_table_refs)/len(exploded):.01%}) inter-table references."
+        )
+        exploded = exploded.loc[
+            exploded.index.difference(
+                inter_table_facts_to_drop.set_index(factoid_idx).index
+            )
+        ].reset_index()
+    else:
+        logger.info("Explode: Found no inter-table references. Dropping nothing.")
+        # reset the index bc our index rn is tbl name and factoid name og.
+        exploded = exploded.reset_index()
+    return exploded
+
+
+def get_table_level(table_name: str, top_table: str) -> int:
+    """Get a table level."""
+    # we may be able to infer this nesting from the metadata
+    table_nesting = {
+        "balance_sheet_assets_ferc1": {
+            "utility_plant_summary_ferc1": {
+                "plant_in_service_ferc1": None,
+                "electric_plant_depreciation_changes_ferc1": None,
+            },
+        },
+        "balance_sheet_liabilities_ferc1": {"retained_earnings_ferc1": None},
+        "income_statement_ferc1": {
+            "depreciation_amortization_summary_ferc1": None,
+            "electric_operating_expenses_ferc1": None,
+            "electric_operating_revenues_ferc1": None,
+        },
+    }
+    if table_name == top_table:
+        level = 1
+    elif table_name in table_nesting[top_table].keys():
+        level = 2
+    elif table_name in pudl.helpers.dedupe_n_flatten_list_of_lists(
+        [values.keys() for values in table_nesting[top_table].values()]
+    ):
+        level = 3
+    else:
+        raise AssertionError(
+            f"AH we didn't find yer table name {table_name} in the nested group of "
+            "tables. Be sure all the tables you are trying to explode are related."
+        )
+    return level
```

Comment:
i expect this whole level getting setup will be nullified if we get [#2625](https://github.com/catalyst-cooperative/pudl/issues/2625) working. i just needed something to work for now